RealmSwift/Combine.swift

```diff
@@ -74,17 +160,86 @@ extension Publisher {
 
 @available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, *)
 extension Publisher where Output: ThreadConfined {
+    /// Enables passing thread-confined objects to a different dispatch queue.
+    ///
+    /// Each call to `receive(on:)` on a publisher which emits Realm
+    /// thread-confined objects must be proceeded by a call to
+    /// `.threadSafeReference()`.The returned publisher handles the required
+    /// logic to pass the thread-confined object to the new queue. Only serial
+    /// dispatch queues are supported and using other schedulers will result in
+    /// a fatal error.
+    ///
+    /// For example, to subscribe on a background thread, do some work there,
+    /// then pass the object to the main thread you can do:
+    ///
+    ///     let cancellable = publisher(myObject)
+    ///         .subscribe(on: DispatchQueue(label: "background queue")
+    ///         .print()
+    ///         .threadSafeReference()
+    ///         .receive(on: DispatchQueue.main)
+    ///         .sink { object in
+    ///             // Do things with the object on the main thread
+    ///         }
+    ///
+    /// Calling this function on a publisher which emits frozen or unmanaged
+    /// objects is unneccesary but is allowed.
+    ///
+    /// - returns: A publisher that supports `receive(on:)` for thread-confined objects.
     public func threadSafeReference() -> Publishers.MakeThreadSafe<Self> {
         Publishers.MakeThreadSafe(self)
     }
 }
 
 @available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, *)
 extension Publisher {
```

Comment:
It might be nice to add a comment to this extension and the above one, giving some context on what functionality is being added and why.